### PR TITLE
Implement MAVLink gimbal v2 protocol

### PR DESCRIPTION
**Pull request by julianoes**
**Describe problem solved by this pull request**

This implements the new MAVLink gimbal v2 protocol according to https://mavlink.io/en/services/gimbal_v2.html.
Note, so far it only implements primary control input, not secondary control input yet which could be used to mix various input sources in the future.

**Describe your solution**
The vmount module now supports the gimbal v2 protocol as an input and/or an output. This means that support for v2 to can be enabled for either the communication to the ground station (`MNT_MODE_IN`) or to a gimbal device (`MNT_MODE_OUT`), or both. This means that legacy/v1 should still be supported where required.

**Test data / coverage**
- This was tested against Auterion Mission Control.
- Unfortunately, at this time, the v2 protocol is not supported by QGroundControl yet.

- MAVSDK needs some updates to support the gimbal v2 manager as implemented here (acquiring control is missing).

**Additional context**
- There is also a [MAVSDK example to test gimbal v2 devices](https://github.com/mavlink/MAVSDK/tree/develop/examples/gimbal_device_tester) which can be used to implement a gimbal v2 device.

Depends on https://github.com/PX4/PX4-SITL_gazebo/pull/703.

### CODE REVIEW

File: src/modules/mavlink/mavlink_bridge_header.h [hunk 1]
```diff
@@ -55,6 +55,12 @@
 #define MAVLINK_GET_CHANNEL_BUFFER mavlink_get_channel_buffer
 #define MAVLINK_GET_CHANNEL_STATUS mavlink_get_channel_status
 
+#if defined(__PX4_POSIX)
```
Comment (dagar) [inline src/modules/mavlink/mavlink_bridge_header.h:58]: Even on NuttX some people want more than 4. 

We have a new `CONSTRAINED_MEMORY` define now (analogous to CONSTRAINED_FLASH) that's only enabled on cortex m4.

Comment (dagar) [inline src/modules/mavlink/mavlink_bridge_header.h:58]:
```suggestion
#ifndef CONSTRAINED_MEMORY
```

File: src/modules/mavlink/mavlink_main.h [hunk 2]
```diff
@@ -540,7 +543,7 @@ class Mavlink : public ModuleParams
 
 	bool			_task_running{true};
 	static bool		_boot_complete;
-	static constexpr int	MAVLINK_MAX_INSTANCES{4};
+	static constexpr int	MAVLINK_MAX_INSTANCES{6};
```
Comment (dagar) [inline src/modules/mavlink/mavlink_main.h:546]: Should it be MAVLINK_COMM_NUM_BUFFERS?

File: src/modules/mavlink/mavlink_params.c [hunk 3]
```diff
@@ -56,7 +56,7 @@ PARAM_DEFINE_INT32(MAV_COMP_ID, 1);
  * @value 1 Always use version 1
  * @value 2 Always use version 2
  */
-PARAM_DEFINE_INT32(MAV_PROTO_VER, 0);
+PARAM_DEFINE_INT32(MAV_PROTO_VER, 2);
```
Comment (dagar) [inline src/modules/mavlink/mavlink_params.c:59]: :+1:

File: src/modules/vmount/input_test.cpp [hunk 4]
```diff
@@ -69,11 +70,16 @@ int InputTest::update(unsigned int timeout_ms, ControlData **control_data, bool
 	_control_data.type_data.angle.frames[1] = ControlData::TypeData::TypeAngle::Frame::AngleAbsoluteFrame;
 	_control_data.type_data.angle.frames[2] = ControlData::TypeData::TypeAngle::Frame::AngleBodyFrame;
 
-	for (int i = 0; i < 3; ++i) {
-		_control_data.type_data.angle.angles[i] = _angles[i] * M_DEG_TO_RAD_F;
+	matrix::Eulerf euler(
+		_angles[0] * M_DEG_TO_RAD_F,
```
Comment (dagar) [inline src/modules/vmount/input_test.cpp:74]: If you use the mathlib instead we're pretty close to not needing the defines. Whichever you prefer.

```suggestion
		math::radians(_angles[0]),
```

File: src/modules/vmount/input_test.cpp [hunk 5]
```diff
@@ -69,11 +70,16 @@ int InputTest::update(unsigned int timeout_ms, ControlData **control_data, bool
...
+		_angles[1] * M_DEG_TO_RAD_F,
+		_angles[2] * M_DEG_TO_RAD_F);
+	matrix::Quatf q(euler);
 
-		_control_data.stabilize_axis[i] = true;
-	}
+	_control_data.type_data.angle.q[0] = q(0);
```
Comment (dagar) [inline src/modules/vmount/input_test.cpp:79]:
```suggestion
	q.copyTo(_control_data.type_data.angle.q);
```

**Comment by dagar** [comment]
Comment out in fmu-v2 to resolve flash shortage.

File: src/modules/mavlink/mavlink_bridge_header.h (hunk 1 above)
Comment (julianoes) [reply src/modules/mavlink/mavlink_bridge_header.h:58]: Nice, I'll change it to that.

File: src/modules/mavlink/mavlink_main.h (hunk 2 above)
Comment (julianoes) [reply src/modules/mavlink/mavlink_main.h:546]: I think you're right.

File: src/modules/vmount/input_test.cpp (hunk 4 above)
Comment (julianoes) [reply src/modules/vmount/input_test.cpp:74]: Thanks, I'm fixing it all over vmount.

File: src/modules/vmount/input_test.cpp (hunk 5 above)
Comment (julianoes) [reply src/modules/vmount/input_test.cpp:79]: Fixed it in two places, thanks.

File: src/modules/mavlink/module.yaml [hunk 6]
```diff
@@ -44,6 +44,8 @@ parameters:
                 #6: Iridium # as the user does not need to configure this, hide it from the UI
                 7: Minimal
                 8: External Vision
+                #9: External Vision Minimal # hidden
```
Comment (dagar) [inline src/modules/mavlink/module.yaml:47]: Why hide this?

**Review by dagar — APPROVED** [review]

**Review by dagar — APPROVED** [review]

File: src/modules/mavlink/module.yaml (hunk 6 above)
Comment (julianoes) [reply src/modules/mavlink/module.yaml:47]: I don't remember, I think that's how it was.

**Comment by julianoes** [comment]
This currently depends on https://github.com/mavlink/mavlink/pull/1578.

**Review by dagar — APPROVED** [review]

**Comment by julianoes** [comment]
@dagar rebased again.

**Review by dagar — APPROVED** [review]

**Pull request merged**